### PR TITLE
bugfix() Sort lookup parameters

### PR DESCRIPTION
**Pull request by simhnna**
Required lookup parameters were not generated at the beginning of the
methods. This caused the generated files to be invalid.
This change makes sure that all required fields are at the beginning of
the parameter list followed by all optional parameters.

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

When generating schema types, the required function parameters were not in the beginning.

## What is the new behavior?
Generated function parameters are now sorted based on them being required and then based on the name.

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

## Other information

### CODE REVIEW

**Comment by kamilmysliwiec** [comment]
Sounds interesting. Thanks for the contribution.

File: lib/graphql-ast.explorer.ts [hunk 1]
```diff
@@ -184,7 +184,21 @@ export class GraphQLAstExplorer {
       returnType: `${type} | Promise<${type}>`,
       parameters: this.getFunctionParameters(
         (item as FieldDefinitionNode).arguments,
-      ),
+      ).sort((a, b) => {
```
Comment (kamilmysliwiec) [inline lib/graphql-ast.explorer.ts]: perhaps we should extract this comparator to a separate function + simplify it a little bit, for example this
```typescript
else if (a.hasQuestionToken) {
    return 1;
} else {
    return -1;
}
```
to:
```typescript
return a.hasQuestionToken ? 1 : -1;
```

Comment (simhnna) [reply lib/graphql-ast.explorer.ts]: Done. I don't have a development environment around now, so I did the changes in github...

File: lib/graphql-ast.explorer.ts [hunk 2]
```diff
@@ -154,6 +154,13 @@ export class GraphQLAstExplorer {
         return this.lookupField(item, parentRef, mode);
     }
   }
+  
+  private sortGraphqlArguments(first: any, second:any): number {
+    if (first.hasQuestionToken && second.hasQuestionToken || !(first.hasQuestionToken || second.hasQuestionToken)) {
+      return first.name === second.name : 0 ? first.name > second.name ? 1 : -1;
```
Comment (Mnwa) [inline lib/graphql-ast.explorer.ts]: I think it's not a good practices two conditional operators in one row

**Pull request closed without merging**